src/iexamine.cpp

```diff
@@ -1555,6 +1555,37 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
     }
 }
 
+/**
+ * Prompt pick cactus pad. Not safe for player.
+ */
+void iexamine::flower_cactus( player &p, const tripoint &examp )
+{
+    if( dead_plant( true, p, examp ) ) {
+        return;
+    }
+
+    if( !query_yn( _( "Pick %s?" ), g->m.furnname( examp ) ) ) {
+        none( p, examp );
+        return;
+    }
+
+    if( one_in( 6 ) ) {
+        add_msg( m_bad, _( "The cactus' nettles sting you!" ) );
+        p.apply_damage( nullptr, bp_arm_l, 4 );
+        p.apply_damage( nullptr, bp_arm_r, 4 );
+    }
+
+    g->m.furn_set( examp, f_null );
+
+    if( p.can_pickWeight( item( "cactus_pad" ), true ) &&
+        p.can_pickVolume( item( "cactus_pad" ), true ) ) {
+        p.i_add( item( "cactus_pad" ) );
+        p.add_msg_if_player( _( "You harvest: cactus pad." ) );
+    } else {
+        g->m.add_item_or_charges( p.pos(), item( "cactus_pad" ) );
```

Comment:
I wouldn't rely on optimizations here: please extract a temp to avoid re-creating the `item`.